docs/testing/runtimes.md

```diff
@@ -0,0 +1,152 @@
+
+- When used for an Excel custom function, the runtime starts up when either the worksheet recalculates or the custom function calculates. It doesn't shut down until the workbook is closed.  
+- When used in an Outlook event-based task, the runtime starts up when the event occurs. It ends when the first of the following occurs:
+
+  - The event handler calls the `completed` method of its event parameter.
+  - 5 minutes has elapsed since the triggering event.
+  - The user changes focus from the window where the event was triggered, such as a message compose window.
+
+A JavaScript-runtime uses less memory and starts up faster than a browser runtime, but has fewer features.
+
+## Browser runtime
+
+Office add-ins use a different browser type runtime depending on the platform in which Office is running (web, Mac, or Windows), and on the version and build of Windows and Office. For example, if the user is running Office on the web in a FireFox browser, then the Firefox runtime is used. If the user is running Office on Mac, then the Safari runtime is used. If the user is running Office on Windows, then either an Edge or Internet Explorer provides the runtime, depending on the version of Windows and Office. Details can be found in [Browsers used by Office Add-ins](../concepts/browsers-used-by-office-web-add-ins.md).
+
+All of these runtimes include an HTML rendering engine and provide support for [WebSockets](https://developer.mozilla.org/docs/Web/API/WebSockets_API), [Full CORS (Cross-Origin Resource Sharing)](https://developer.mozilla.org/docs/Web/HTTP/CORS), and [local storage](https://developer.mozilla.org/docs/Web/API/Window/localStorage), and cookies. 
+
+A browser runtime lifespan varies depending on the feature that it implements and on whether it is being shared or not.
+
+- When an add-in with a task pane is launched, a browser runtime starts, unless it is a shared runtime that is already running. If it is a shared runtime it shuts down when the document is closed. If it is an unshared runtime, it shuts down when the task pane is closed.
+- When a dialog is opened, a browser runtime starts. It shuts down when the dialog is closed.
+- When a function command is executed (which happens when a user selects its button or menu item), a browser runtime starts, unless it is a shared runtime that is already running. If it is a shared runtime it shuts down when the document is closed. If it is an unshared runtime, it shuts down when the first of the following occurs:
+ 
+  - The function command calls the `completed` method of its event parameter.
+  - 5 minutes has elapsed since the triggering event. (If a dialog was opened in the function command and it is still open when the parent runtime times-out, the dialog runtime stays running until the dialog is closed.)
```

Comment:
```suggestion
  - 5 minutes have elapsed since the triggering event. (If a dialog was opened in the function command and it's still open when the parent runtime times out, the dialog runtime stays running until the dialog is closed.)
```